setuptools/__init__.py

```diff
@@ -167,6 +167,12 @@ class Command(_Command):
     command_consumes_arguments = False
     distribution: Distribution  # override distutils.dist.Distribution with setuptools.dist.Distribution
 
+    dry_run = False
```

Comment:
stdlib distutils types this using an old `Literal[0, 1]` instead of a boolean.
This'll fix itself with https://github.com/pypa/setuptools/issues/4689
In the mean time you can either type/pyright ignore, or use `0`

Reply:
```suggestion
    dry_run = False  # pyright: ignore # (until #4689; see #4872)
```

Reply:
Quick update: https://github.com/python/typeshed/pull/13615 got merged in typeshed
So next version of mypy that includes a typeshed sync should also resolve this.